### PR TITLE
Refactors Spell and Martial Art Books to use Action Granters!

### PR DESCRIPTION
**Pull request by ghost**
Action granting books are way nicer code wise than a child of the wizard's buying book, the pages the mastery vars reading sounds and all that are way better and it finally makes use of a fleshed out reading system.

SOME THINGS YOU WILL NOTICE:
1. spell books take time to read
2. spell books still recoil, but the user will decide not to read if they already have the spell (i want to keep this because it stops the chaplain thing where they can spam their lesser smoke book to be infinitely hungry and eat the entire chef's wares BUT I can remove this if you want or just make only wizards recognize their superiority)
3. spell books can now be set to be infinite use, for admins!! yay!!
4. because i'm adding all the new spells that lazy kor coders didn't add to spellbooks, they will show up now.
5. I'm probably going to make the mimery books into action granting so i can remove that dumb blacklist check. they are just two actions anyways, why not?

TODO:
~~remove the old spellbook oneuse code~~
~~add the new spells~~ (this didn't happen because the spells are all robed, i don't know why i didn't think of that sorry)
~~hell add the old spells~~
~~move mimery into an action~~
~~PROBABLY move these over into a new file~~
port martial arts

Lich: NO
Link Worlds: Admin only and I don't want to balance in my refactor pr

### CODE REVIEW

**Comment by Qustinnus** [comment]
nice commit name, also nice travis

File: code/game/objects/items/manuals.dm [hunk 1]
```diff
@@ -958,15 +958,17 @@
 	title = "Hacking"
 	page_link = "Hacking"
 
-///Reusable books that grant actions (knowledge is power)
-/obj/item/book/action_granting
```
Comment (Qustinnus) [inline code/game/objects/items/manuals.dm]: you didn't change the name of these typepaths in places like cargo

Comment (Armhulen) [reply code/game/objects/items/manuals.dm]: fuck cargo i'm going off the grid (okay i'll fix it later thank you)

Comment (Armhulen) [reply code/game/objects/items/manuals.dm]: done!

**Comment by Armhulen** [comment]
I need to repath lavaland loot!!

**Comment by Armhulen** [comment]
EASY

File: code/game/objects/items/manuals.dm [hunk 2]
```diff
@@ -1013,4 +1022,213 @@
...
+	desc = "This book is more horse than your mind has room for."
+	remarks = list("Moooooooo!","Moo!","Moooo!", "NEEIIGGGHHHH!", "NEEEIIIIGHH!", "NEIIIGGHH!", "HAAWWWWW!", "HAAAWWW!", "Oink!", "Squeeeeeeee!", "Oink Oink!", "Ree!!", "Reee!!", "REEE!!", "REEEEE!!")
+
+/obj/item/book/granter/spell/barnyard/recoil(mob/living/carbon/user)
+	if(ishuman(user))
+		to_chat(user,"<font size='15' color='red'><b>HOR-SIE HAS RISEN</b></font>")
+		var/obj/item/clothing/mask/horsehead/magichead = new /obj/item/clothing/mask/horsehead
+		magichead.flags_1 |= NODROP_1		//curses!
+		magichead.flags_inv &= ~HIDEFACE //so you can still see their face
+		magichead.voicechange = 1	//NEEEEIIGHH
```
Comment (ShizCalev) [inline code/game/objects/items/manuals.dm]: TRUE

File: code/game/objects/items/manuals.dm [hunk 3]
```diff
@@ -1013,4 +1022,213 @@
...
+		if(!user.dropItemToGround(user.wear_mask))
+			qdel(user.wear_mask)
+		user.equip_to_slot_if_possible(magichead, slot_wear_mask, 1, 1)
```
Comment (ShizCalev) [inline code/game/objects/items/manuals.dm]: true, true

File: code/modules/spells/spell_types/charge.dm [hunk 4]
```diff
@@ -34,19 +34,22 @@
...
+				var/obj/item/book/granter/spell/I = item
+				if(!I.oneuse)
+					to_chat(L, "<span class='notice'>This book is infinite use and can't be recharged, yet the magic has improved the book somehow...</span>")
 					burnt_out = 1
+					I.pages_to_mastery--
+				if(prob(80))
+					L.visible_message("<span class='warning'>[I] catches fire!</span>")
+					qdel(I)
+				else
+					I.used = 0
```
Comment (ShizCalev) [inline code/modules/spells/spell_types/charge.dm]: false

File: code/modules/spells/spell_types/charge.dm [hunk 5]
```diff
@@ -34,19 +34,22 @@
...
-					else
-						I.used = 0
-						charged_item = I
-						break
-				else
-					to_chat(L, "<span class='danger'>Glowing red letters appear on the front cover...</span>")
-					to_chat(L, "<span class='warning'>[pick("NICE TRY BUT NO!","CLEVER BUT NOT CLEVER ENOUGH!", "SUCH FLAGRANT CHEESING IS WHY WE ACCEPTED YOUR APPLICATION!", "CUTE! VERY CUTE!", "YOU DIDN'T THINK IT'D BE THAT EASY, DID YOU?")]</span>")
+				to_chat(L, "<span class='danger'>Glowing red letters appear on the front cover...</span>")
+				to_chat(L, "<span class='warning'>[pick("NICE TRY BUT NO!","CLEVER BUT NOT CLEVER ENOUGH!", "SUCH FLAGRANT CHEESING IS WHY WE ACCEPTED YOUR APPLICATION!", "CUTE! VERY CUTE!", "YOU DIDN'T THINK IT'D BE THAT EASY, DID YOU?")]</span>")
+				burnt_out = 1
```
Comment (ShizCalev) [inline code/modules/spells/spell_types/charge.dm]: true

File: code/modules/spells/spell_types/charge.dm [hunk 6]
```diff
@@ -34,19 +34,22 @@
...
+			else if(istype(item, /obj/item/book/granter/spell))
+				var/obj/item/book/granter/spell/I = item
+				if(!I.oneuse)
+					to_chat(L, "<span class='notice'>This book is infinite use and can't be recharged, yet the magic has improved the book somehow...</span>")
 					burnt_out = 1
```
Comment (ShizCalev) [inline code/modules/spells/spell_types/charge.dm]: true

File: code/game/objects/items/manuals.dm [hunk 7]
```diff
@@ -1013,4 +1022,213 @@
...
+/obj/item/book/granter/spell/fireball
+	spell = /obj/effect/proc_holder/spell/aimed/fireball
+	spellname = "fireball"
+	icon_state ="bookfireball"
+	desc = "This book feels warm to the touch."
+	remarks = list("Aim...AIM, FOOL!", "Just catching them on fire won't do...", "Accounting for crosswinds... really?", "I think I just burned my hand...", "Why the dumb stance? It's just a flick of the hand...", "OMEE... ONI... Ugh...", "What's the difference between a fireball and a pyroblast...")
+
+/obj/item/book/granter/spell/fireball/recoil(mob/user)
+	..()
+	explosion(user.loc, -1, 0, 2, 3, 0, flame_range = 2)
```
Comment (ShizCalev) [inline code/game/objects/items/manuals.dm]: explosion(user.loc, -1, 0, 2, 3, FALSE, FALSE, 2)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/game/objects/items/manuals.dm (hunk 7 above)
Comment (ShizCalev) [reply code/game/objects/items/manuals.dm]: though that said, is -1 devastation_range intended?

Comment (Armhulen) [reply code/game/objects/items/manuals.dm]: Just doing what the old spell book did let's not do balance yes?

**Comment by Cyberboss** [comment]
@MapDiffBot

File: code/game/objects/items/manuals.dm [hunk 8]
```diff
@@ -1013,4 +1022,213 @@
...
+	if(!stored_swap)
+		stored_swap = user
+		to_chat(user,"<span class='warning'>For a moment you feel like you don't even know who you are anymore.</span>")
+		return
+	if(stored_swap == user)
+		to_chat(user,"<span class='notice'>You stare at the book some more, but there doesn't seem to be anything else to learn...</span>")
+		return
+
+	var/obj/effect/proc_holder/spell/targeted/mind_transfer/swapper = new
+	swapper.cast(user, stored_swap, TRUE
```
Comment (ShizCalev) [inline code/game/objects/items/manuals.dm]: closing

Comment (Armhulen) [reply code/game/objects/items/manuals.dm]: woop

**Comment by ShizCalev** [comment]
compile

**Comment by Armhulen** [comment]
actually let me see if i can make this title less retarded

**Comment by Cyberboss** [comment]
@MapDiffBot

**Comment by Cyberboss** [comment]
@MapDiffBot

**Comment by Armhulen** [comment]
@ShizCalev

**Comment by ShizCalev** [comment]
doesn't compile @Armhulen

**Comment by Armhulen** [comment]
?!

i'll fix it one second

**Comment by Armhulen** [comment]
@ShizCalev sssssssssssssssssssssssssssssssssshoooould be good...

unless i'm forgetting anything this is ready

**Pull request merged**